shane_fisher/rails2/DojoSecrets/app/controllers/secrets_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SecretsController < ApplicationController
+  def index
+  end
+
+  def new
+    @secrets = Secret.all
+  end
+
+  def create
+    @secret=Secret.create(content: params[:Content], user_id: current_user.id)
+    redirect_to :back
+  end
+
+  def destroy
+    Secret.find(params[:id]).destroy
+    redirect_to :back
+  end
+
+  def like
+    @secret = Secret.find(params[:id])
+    if params[:like] == "Unlike" 
```

Comment:
Very interesting to pass the param for like or unlike. +1 for thinking outside the box and having one method to handle both cases. I bet you could simplify this with the same concept in mind, how else would we know if the user should be unliking the secret? Hint: if the relationship exists